src/main/java/mate/academy/spring/model/User.java

```diff
@@ -41,10 +50,36 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    public Set<Role> getRoles() {
+        return roles;
+    }
+
+    public void setRoles(Set<Role> roles) {
+        this.roles = roles;
+    }
+
     @Override
     public String toString() {
         return "User{"
                 + "id=" + id
                 + ", email='" + email + '\'' + '}';
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        User user = (User) o;
+        return Objects.equals(id, user.id) && Objects.equals(email, user.email)
+                && Objects.equals(password, user.password) && Objects.equals(roles, user.roles);
```

Comment:
we shouldn't use user confidential credentials anywhere outside of places absolutely necessary for authentication logic